File: fe2-android/app/src/main/java/com/github/dedis/popstellar/model/network/method/message/data/consensus/ConsensusPrepare.java
package com.github.dedis.popstellar.model.network.method.message.data.consensus;

import com.github.dedis.popstellar.model.network.method.message.data.*;
import com.github.dedis.popstellar.model.objects.security.MessageID;
import com.google.gson.annotations.SerializedName;

public final class ConsensusPrepare extends Data {

  @SerializedName("instance_id")
  private final String instanceId;

  @SerializedName("message_id")
  private final MessageID messageId;

  @SerializedName("created_at")
  private final long creation;

  @SerializedName("value")
  private final PrepareValue prepareValue;

  /**
   * Constructor for a data Prepare
   *
   * @param instanceId unique id of the consensus instance
   * @param messageId message id of the Elect message
   * @param creation UNIX timestamp in UTC
   * @param proposedTry proposed try number
   */
  public ConsensusPrepare(String instanceId, MessageID messageId, long creation, int proposedTry) {
    this.instanceId = instanceId;
    this.messageId = messageId;
    this.creation = creation;
    this.prepareValue = new PrepareValue(proposedTry);
  }

  @Override
  public String getObject() {
    return Objects.CONSENSUS.getObject();
  }

  @Override
  public String getAction() {
    return Action.PREPARE.getAction();
  }

  public String getInstanceId() {
    return instanceId;
  }

  public MessageID getMessageId() {
    return messageId;
  }

  public long getCreation() {
    return creation;
  }

  public PrepareValue getPrepareValue() {
    return prepareValue;
  }

  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    ConsensusPrepare that = (ConsensusPrepare) o;

    return creation == that.creation
        && java.util.Objects.equals(instanceId, that.instanceId)
        && java.util.Objects.equals(messageId, that.messageId)
        && java.util.Objects.equals(prepareValue, that.prepareValue);
  }

  @Override
  public int hashCode() {
    return java.util.Objects.hash(instanceId, messageId, creation, prepareValue);
  }

  @Override
  public String toString() {
    return String.format(
        "ConsensusPrepare{instance_id='%s', message_id='%s', created_at=%s, value=%s}",
        instanceId, messageId.getEncoded(), creation, prepareValue);
  }
}

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/model/network/method/message/data/consensus/ConsensusPrepare.kt
package com.github.dedis.popstellar.model.network.method.message.data.consensus

import com.github.dedis.popstellar.model.Immutable
import com.github.dedis.popstellar.model.network.method.message.data.Action
import com.github.dedis.popstellar.model.network.method.message.data.Data
import com.github.dedis.popstellar.model.network.method.message.data.Objects
import com.github.dedis.popstellar.model.objects.security.MessageID
import com.google.gson.annotations.SerializedName

@Immutable
class ConsensusPrepare(
    @field:SerializedName("instance_id") val instanceId: String,
    @field:SerializedName("message_id") val messageId: MessageID,
    @field:SerializedName("created_at") val creation: Long,
    proposedTry: Int
) : Data {

  @SerializedName("value") val prepareValue: PrepareValue

  /**
   * Constructor for a data Prepare
   *
   * @param instanceId unique id of the consensus instance
   * @param messageId message id of the Elect message
   * @param creation UNIX timestamp in UTC
   * @param proposedTry proposed try number
   */
  init {
    prepareValue = PrepareValue(proposedTry)
  }

  override val `object`: String = Objects.CONSENSUS.`object`
  override val action: String = Action.PREPARE.action

  override fun equals(other: Any?): Boolean {
    if (this === other) {
      return true
    }
    if (other == null || javaClass != other.javaClass) {
      return false
    }
    val that = other as ConsensusPrepare
    return creation == that.creation &&
        instanceId == that.instanceId &&
        messageId == that.messageId &&
        prepareValue == that.prepareValue
  }

  override fun hashCode(): Int {
    return java.util.Objects.hash(instanceId, messageId, creation, prepareValue)
  }

  override fun toString(): String {
    return "ConsensusPrepare{instance_id='$instanceId', message_id='${messageId.encoded}', " +
        "created_at=$creation, value=$prepareValue}"
  }
}
